### PR TITLE
Support non-HA mode

### PR DESCRIPTION
**Pull request by ravidbro**

### CODE REVIEW

**Comment by sttts** [comment]
SGTM, after api has merged and the bump is real here.

File: go.mod [hunk 1]
```diff
@@ -13,19 +13,21 @@ require (
...
+	golang.org/x/net v0.0.0-20201110031124-69a78807bb2b
 	gopkg.in/square/go-jose.v2 v2.2.2
 	gopkg.in/yaml.v2 v2.3.0
-	k8s.io/api v0.19.2
-	k8s.io/apimachinery v0.19.2
-	k8s.io/apiserver v0.19.2
-	k8s.io/client-go v0.19.2
-	k8s.io/component-base v0.19.2
-	k8s.io/klog/v2 v2.3.0
+	k8s.io/api v0.20.0
```
Comment (stlaz) [inline go.mod:22]: this does way more than what the commit is saying in its commit message. Please mention the kube bump as well.

File: pkg/controllers/readiness/unsupported_override.go [hunk 2]
```diff
@@ -3,6 +3,7 @@ package readiness
 import (
 	"bytes"
 	"encoding/json"
+	v1 "github.com/openshift/api/config/v1"
```
Comment (stlaz) [inline pkg/controllers/readiness/unsupported_override.go:6]: configv1

File: pkg/controllers/readiness/unsupported_override.go [hunk 3]
```diff
@@ -54,7 +55,11 @@ func isUnsupportedUnsafeAuthentication(spec *operatorv1.OperatorSpec) (bool, err
 	}
 }
 
-func getExpectedMinimumNumberOfMasters(spec *operatorv1.OperatorSpec) int {
+func getExpectedMinimumNumberOfMasters(spec *operatorv1.OperatorSpec, clusterHAMode v1.HighAvailabilityMode) int {
+	if clusterHAMode == v1.NoneHighAvailabilityMode {
+		return 1
+	}
```
Comment (stlaz) [inline pkg/controllers/readiness/unsupported_override.go:61]: please incorporate this in the switch below

**Review by stlaz — CHANGES_REQUESTED** [review]
mostly just nits and conventions

File: go.mod (hunk 1 above)
Comment (ravidbro) [reply go.mod:22]: done

File: pkg/controllers/readiness/unsupported_override.go (hunk 2 above)
Comment (ravidbro) [reply pkg/controllers/readiness/unsupported_override.go]: done

**Comment by stlaz** [comment]
@ravidbro This is going to require an e2e test, feel free to open that as a follow-up.

**Comment by ravidbro** [comment]
> @ravidbro This is going to require an e2e test, feel free to open that as a follow-up.

AFAIK, @osherdp is working on the e2e test for SNO

**Comment by osherdp** [comment]
@ravidbro that's right

File: pkg/controllers/readiness/unsupported_override.go [hunk 4]
```diff
@@ -54,14 +55,16 @@ func isUnsupportedUnsafeAuthentication(spec *operatorv1.OperatorSpec) (bool, err
 	}
 }
 
-func getExpectedMinimumNumberOfMasters(spec *operatorv1.OperatorSpec) int {
+func getExpectedMinimumNumberOfMasters(spec *operatorv1.OperatorSpec, topologyMode configv1.TopologyMode) int {
 	allowAnyNumber, err := isUnsupportedUnsafeAuthentication(spec)
 	switch {
 	case err != nil:
 		utilruntime.HandleError(err)
 		return 3
```
Comment (dhellmann) [inline pkg/controllers/readiness/unsupported_override.go:63]: Shouldn't the topology mode override this default of 3 as well?

Comment (ravidbro) [reply pkg/controllers/readiness/unsupported_override.go:65]: fixed

**Pull request merged**